src/endpoint/devices.ts

```diff
@@ -205,6 +205,20 @@ export interface CommandList {
 	commands: Command[]
 }
 
+export type PreferenceType = 'integer' | 'number' | 'boolean' | 'string' | 'enumeration'
+
+export interface DevicePreferenceEntry {
+	preferenceType: PreferenceType
+	value: string | number | boolean
+}
+
+export interface DevicePreferenceResponse {
+	/**
+	 * Map of PreferenceId to its stored values
+	 */
+	values?: { [key: string]: DevicePreferenceEntry}
```

Comment:
We could use a `Record` here instead, and based on the possible values of `PreferenceId`, we would want to use a `Partial<Record<string, DevicePreferenceEntry>>` as I learned [here](https://fnune.com/typescript/2019/01/30/typescript-series-1-record-is-usually-not-the-best-choice/).